src/library/searchqueryparser.cpp

```diff
@@ -104,51 +144,18 @@ QString SearchQueryParser::getTextArgument(QString argument,
             argument = tokens->takeFirst();
         }
     }
-
-    bool shouldMatchExactly = false;
+    StringMatch mode = StringMatch::Contains;
     if (argument.startsWith("=")) {
+        // strip the '=' from the argument
         argument = argument.mid(1);
-        shouldMatchExactly = true;
+        mode = StringMatch::Equals;
     }
-    // Deal with quoted arguments. If this token started with a quote, then
-    // search for the closing quote.
     if (argument.startsWith("\"")) {
-        argument = argument.mid(1);
-
-        int quote_index = argument.indexOf("\"");
-        while (quote_index == -1 && tokens->length() > 0) {
-            argument += " " + tokens->takeFirst();
-            quote_index = argument.indexOf("\"");
-        }
-
-        if (quote_index == -1) {
-            // No ending quote found. Since we think they are going to close the
-            // quote eventually, treat the entire token list as the argument for
-            // now.
-            return argument;
-        }
-
-        // Stuff the rest of the argument after the quote back into tokens.
-        QString remaining = argument.mid(quote_index+1).trimmed();
-        if (remaining.size() != 0) {
-            tokens->push_front(remaining);
-        }
-
-        if (quote_index == 0) {
-            // We have found an explicit empty string ""
-            // return it as "" to distinguish it from an unfinished empty string
-            argument = kMissingFieldSearchTerm;
-        } else {
-            // Found a closing quote.
-            // Slice off the quote and everything after.
-            argument = argument.left(quote_index);
-            if (matchMode != nullptr && shouldMatchExactly) {
-                *matchMode = StringMatch::Equals;
-            }
-        }
+        Quoted quoted;
+        std::tie(argument, quoted) = consumeQuotedArgument(argument, tokens);
```

Comment:
magic...

Reply:
too much?

Reply:
nope. just had to look up what std::tie actually does.